apps/src/lib/node/ledger/events/log/dumb_queries.rs

```diff
@@ -6,118 +6,71 @@
 //! tm.event='NewBlock' AND <accepted|applied>.<$attr>='<$value>'
 //! ```
 
-use lazy_static::lazy_static;
-use regex::Regex;
+use namada::types::hash::Hash;
 
 use crate::node::ledger::events::{Event, EventType};
 
-/// Regular expression used to parse Tendermint queries.
-const QUERY_PARSING_REGEX_STR: &str =
-    r"^tm\.event='NewBlock' AND (accepted|applied)\.([\w_]+)='([^']+)'$";
-
-lazy_static! {
-    /// Compiled regular expression used to parse Tendermint queries.
-    static ref QUERY_PARSING_REGEX: Regex = Regex::new(QUERY_PARSING_REGEX_STR).unwrap();
-}
-
 /// A [`QueryMatcher`] verifies if a Namada event matches a
 /// given Tendermint query.
 #[derive(Debug, Clone)]
-pub struct QueryMatcher<'q> {
+pub struct QueryMatcher {
     event_type: EventType,
     attr: String,
-    value: &'q str,
+    value: Hash,
 }
 
-impl<'q> QueryMatcher<'q> {
+impl QueryMatcher {
     /// Checks if this [`QueryMatcher`] validates the
     /// given [`Event`].
     pub fn matches(&self, event: &Event) -> bool {
         event.event_type == self.event_type
             && event
                 .attributes
                 .get(&self.attr)
-                .map(|value| value == self.value)
+                .and_then(|value| {
+                    value
+                        .as_str()
+                        .try_into()
```

Comment:
at the rpc level we already go from a hex encoded hash to a regular hash, and perform sanity checks at that level. this just does a roundtrip encoding, to compare with the hash in the query.

Reply:
From the perspective of this fn this isn't infallible though, it could be passed an `Event` that could not have a hex hash at the `attr`. I guess `Event::attributes` should be something more typed than `HashMap<String, String>`.

More generally than this PR I want to look at how we are handling errors in our code, I strongly think we should handle all `Err`s even if they should never occur, in case one of our underlying assumptions is wrong (or becomes invalid later on), or the function comes to be used somewhere else where our assumption why some error will never occur doesn't hold. So imo we should log or refactor such there is no `Err` to deal with.